Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -0,0 +1,300 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+class ListMenu_MenuInfo
+{
+  constructor(public readonly ID: string, public readonly ItemInfos: ListMenuItemInfo[])
+  {
+  }
+}
+
+class ListMenuItemInfo
+{
+  constructor(
+    public readonly ID: string,
+    public readonly Category: string,
+    public readonly Text: string,
+    public readonly Icon: string,
+    public readonly IconDisabled: string,
+    public readonly RequiredSelection: number,
+    public readonly IsDisabled: boolean,
+    public readonly Href: string,
+    public readonly Target: string,
+    public readonly DiagnosticMetadata: Dictionary<string | boolean>,
+    public readonly DiagnosticMetadataForCommand: Dictionary<string | boolean>)
+  {
+  }
+}
+
+class ListMenu
+{
+  private static _listMenuInfos: Dictionary<ListMenu_MenuInfo> = {};
+  private static _itemClassName: string = 'listMenuItem';
```

Comment:
This kind of thing should be updated to const when we do the cleanup. please add note to follow-up task for this. @a-ctor  known where / how

Reply:
Class members cannot be `const`, only function variables can be declared `const`. Or do you mean `readonly`? We do have a task to change `var`s into `let`s/`const`s but no task for `readonly` fields

Reply:
{{readonly}} it is, then. Yes, please add a task for that.

Reply:
@MichaelKetting can I add the `readonly` refactoring to the `let`/`const` task or do you want to create a separate task for it?

Reply:
add it. should fit in nicely.